### PR TITLE
MGMT-18829: Convert agent-install CRDs to new webhook

### PR DESCRIPTION
**Pull request by CrystalChun**
Moves the webhook logic for the agent-install CRDs (Agent, InfraEnv, and AgentClassification) from [pkg/webhooks](https://github.com/openshift/assisted-service/tree/master/pkg/webhooks) and scaffolds the new webhook files.

---

First PR in multi-part change to convert the old webhooks using the Kube API aggregate server to the new webhooks. This change does NOT start using the new webhooks yet.

---
Motivation for this change is to prevent us being closely coupled to the Kubernetes version running our service. See https://github.com/kubernetes/kubernetes/issues/124533 and [MGMT-18829](https://issues.redhat.com/browse/MGMT-18829) for more details.

## List all the issues related to this PR

- [ ] New Feature 
- [x] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] This change does not require a documentation update (docstring, `docs`, README, etc): Doc update will come once the webhooks are in use
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

/cc @carbonin

### CODE REVIEW

**Comment by CrystalChun** [comment]
After rebasing from master it's failing to build with
```
+ go install github.com/axw/gocov/gocov@latest
go: downloading github.com/axw/gocov v1.2.1
go: github.com/axw/gocov/gocov@latest: github.com/axw/gocov@v1.2.1 requires go >= 1.22.0 (running go 1.21.13; GOTOOLCHAIN=local)
```

File: api/v1beta1/agent_webhook.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+var agentlog = logf.Log.WithName("agent-resource")
+
+// SetupWebhookWithManager will setup the manager to manage the webhooks
+func (r *Agent) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+//+kubebuilder:webhook:path=/validate-agent-install-openshift-io-v1beta1-agent,mutating=false,failurePolicy=fail,sideEffects=None,groups=agent-install.openshift.io,resources=agents,verbs=create;update,versions=v1beta1,name=vagent.kb.io,admissionReviewVersions=v1
```
Comment (jhernand) [inline api/v1beta1/agent_webhook.go:41]: Is the `create` verb needed here if we are only validating updates?

File: api/v1beta1/infraenv_webhook.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+var infraenvlog = logf.Log.WithName("infraenv-resource")
+
+// SetupWebhookWithManager will setup the manager to manage the webhooks
+func (r *InfraEnv) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+//+kubebuilder:webhook:path=/validate-agent-install-openshift-io-v1beta1-infraenv,mutating=false,failurePolicy=fail,sideEffects=None,groups=agent-install.openshift.io,resources=infraenvs,verbs=create;update,versions=v1beta1,name=vinfraenv.kb.io,admissionReviewVersions=v1
```
Comment (jhernand) [inline api/v1beta1/infraenv_webhook.go:39]: Is the `create` verb needed here?

**Comment by jhernand** [comment]
This looks good to me in general. The only thing that I dislike is that the implementations of the webhooks live in the "api" package. If would be great if we could move that to some other package. I understand that may be impossible because this code was originally generated with the Operator SDK. The reason I don't like that is that it makes it more complicated for other projects to import the API of this project, although in think we don't really expect or support that.

File: api/v1beta1/agent_webhook.go (hunk 1 above)
Comment (CrystalChun) [reply api/v1beta1/agent_webhook.go:41]: I found out that this is needed for the Infrastructure Operator to create the `ValidatingWebhookConfiguration` CR

File: api/v1beta1/infraenv_webhook.go (hunk 2 above)
Comment (CrystalChun) [reply api/v1beta1/infraenv_webhook.go:39]: I found out that this is needed for the Infrastructure Operator to create the `ValidatingWebhookConfiguration` CR

**Pull request merged**